replica/database.cc

```diff
@@ -340,7 +340,7 @@ database::database(const db::config& cfg, database_config dbcfg, service::migrat
     // No timeouts or queue length limits - a failure here can kill an entire repair.
     // Trust the caller to limit concurrency.
     , _streaming_concurrency_sem(
-            max_count_streaming_concurrent_reads,
+            _cfg.maintenance_reader_concurrency_semaphore_count_limit,
```

Comment:
That's not going to work, see #7316 , the _cfg lives on shard-0 and binding it to shard-N is broken :(

Reply:
Oh, wait, maybe it will work. Copying is broken, but you bind it in the correct shard already